### PR TITLE
warn on implicit use of log scale in `categorical`, `bernoulli`

### PR DESCRIPTION
**Pull request by littleredcomputer**
- add a wrapper which can be used to issue a deprecation warning for the use of bare parameters in those cases where we implicitly selected log scale for probability.

- Fixed docs, unit tests and examples

- fixed string repr of distributions to map camel to snake case, corresponding to GenJAX usage instead of TF

### CODE REVIEW

File: src/genjax/_src/generative_functions/distributions/distribution.py [hunk 1]
```diff
@@ -420,10 +421,22 @@ def _check():
                 return w, v
 
 
+def canonicalize_distribution_name(s) -> str:
+    """Converts underlyinig distribution name from CamelCase to snake_case
```
Comment (sritchie) [inline src/genjax/_src/generative_functions/distributions/distribution.py]: typo on underlyinig

File: src/genjax/_src/generative_functions/distributions/distribution.py [hunk 2]
```diff
@@ -420,10 +421,22 @@ def _check():
...
+    and prepents `genjax.`"""
```
Comment (sritchie) [inline src/genjax/_src/generative_functions/distributions/distribution.py]: "prepends"

File: src/genjax/_src/generative_functions/distributions/distribution.py [hunk 3]
```diff
@@ -420,10 +421,22 @@ def _check():
                 return w, v
 
 
+def canonicalize_distribution_name(s) -> str:
```
Comment (sritchie) [inline src/genjax/_src/generative_functions/distributions/distribution.py]: add `s: str`?

File: docs/cookbook/inactive/expressivity/ravi_stack.ipynb [hunk 4]
```diff
@@ -44,7 +44,7 @@
    "source": [
     "@gen\n",
     "def model():\n",
-    "    idx = genjax.categorical([0.5, 0.25, 0.25]) @ \"idx\"\n",
```
Comment (sritchie) [inline docs/cookbook/inactive/expressivity/ravi_stack.ipynb:47]: so was this a case of someone thinking they were passing probs when in fact they were passing logits?

Comment (sritchie) [reply docs/cookbook/inactive/expressivity/ravi_stack.ipynb:47]: Looks like it!!

File: tests/generative_functions/test_static_gen_fn.py [hunk 5]
```diff
@@ -147,7 +147,7 @@ def test_get_zero_trace(self):
         @genjax.gen
         def model(x):
             y = genjax.normal(x, 1.0) @ "y"
-            z = genjax.bernoulli(0.7) @ "z"
```
Comment (sritchie) [inline tests/generative_functions/test_static_gen_fn.py:150]: wow, basically in all cases this was probably a mistake.

**Review by sritchie — APPROVED** [review]
